homeassistant/helpers/template.py

```diff
@@ -571,46 +574,105 @@ def __init__(self, hass, state):
         self._hass = hass
         self._state = state
 
-    def _access_state(self):
-        state = object.__getattribute__(self, "_state")
-        hass = object.__getattribute__(self, "_hass")
-        _collect_state(hass, state.entity_id)
-        return state
+    def _collect_state(self):
+        if _RENDER_INFO in self._hass.data:
+            self._hass.data[_RENDER_INFO].entities.add(self._state.entity_id)
+
+    # Jinja will try __getitem__ first and it avoids the need
+    # to call is_safe_attribute
+    def __getitem__(self, item):
+        """Return a property as an attribute for jinja."""
+        if item in [
+            "state",
+            "attributes",
+            "last_changed",
+            "last_updated",
+            "context",
+            "domain",
+            "object_id",
+            "name",
+        ]:
```

Comment:
Would tuple be faster here? 

```suggestion
        if item in (
            "state",
            "attributes",
            "last_changed",
            "last_updated",
            "context",
            "domain",
            "object_id",
            "name",
        ):
```

Reply:
Yes, for sure.

Reply:
Thinking about it some more, I should probably make it a set since that will be O(1)

Reply:
done